syntax = "proto3";

package protomsg;

service Detector{
    rpc detect_img(Image) returns (Detection) {}
}

service Tester {
    rpc test_grpc(TestMsg) returns (TestMsg) {}
}

message TestMsg {
    int32 x = 1;
}

message Image {
    int32 width = 1;
    int32 height = 2;
    bytes data = 3;
}

message Object {
    string label = 1;
    float probability = 2;
    int32 xmin = 3;
    int32 ymin = 4;
    int32 xmax = 5;
    int32 ymax = 6;
}

message Detection {
    repeated Object objects = 1;
}

// python -m grpc_tools.protoc -I. --python_out=. --grpc_python_out=. ./msg.proto
// protoc -I . --csharp_out=client --grpc_out=client --plugin=protoc-gen-grpc="C:\Users\cwc5421\Downloads\grpc.tools.2.26.0\tools\windows_x64\grpc_csharp_plugin.exe" msg.proto
// protoc -I . --csharp_out=client --grpc_out=client --plugin=protoc-gen-grpc="C:\Users\cwc5421\Downloads\Grpc.Tools.2.26.0-dev201912021138\tools\windows_x64\grpc_csharp_plugin.exe" msg.proto